common-context/createCommonContextFunctions.js

```diff
@@ -51,58 +49,63 @@ const resolveUserAccessRights = (member, round, courseCode, semester) => {
 }
 
 // --- Building up courseTitle, courseData, semesters and roundData and check access for rounds ---//
-function handleCourseData(courseObject, courseCode, userName, language) {
-  if (!courseObject) {
-    this.errorMessage = 'Whoopsi daisy... kan just nu inte hämta data från kopps'
+function handleCourseData(courseData, courseCode) {
+  const { ladokCourseRounds, ladokData: ladokCourseObject } = courseData
+  if (!ladokCourseObject) {
+    this.errorMessage = 'Whoopsi daisy... kan just nu inte hämta data från Ladok'
     return
   }
   try {
-    const { course, formattedGradeScales, termsWithCourseRounds } = courseObject
+    const { benamning, omfattning } = ladokCourseObject
 
     this.courseData = {
       courseCode,
-      gradeScale: formattedGradeScales,
       semesterObjectList: {},
     }
     this.courseTitle = {
-      name: course.title[this.language === 0 ? 'en' : 'sv'],
-      credits: course.credits.toString().indexOf('.') < 0 ? course.credits + '.0' : course.credits,
+      name: benamning,
+      credits: omfattning,
     }
 
-    for (let semesterIndex = 0; semesterIndex < courseObject.termsWithCourseRounds.length; semesterIndex++) {
-      const { term: roundSemester, rounds } = termsWithCourseRounds[semesterIndex]
+    const thisStore = this
 
-      this.courseData.semesterObjectList[roundSemester] = {
-        rounds,
+    const groupedLadokCourseRounds = []
+    ladokCourseRounds.forEach(round => {
+      if (groupedLadokCourseRounds.length > 0) {
+        groupedLadokCourseRounds.forEach(group => {
+          if (group.term == round.startperiod.inDigits) {
+            group.rounds.push(round)
+          } else {
+            groupedLadokCourseRounds.push({ term: round.startperiod.inDigits, rounds: [round] })
+          }
+        })
+      } else {
+        groupedLadokCourseRounds.push({ term: round.startperiod.inDigits, rounds: [round] })
       }
-    }
+    })
```

Comment:
I think it can be good to create a few unit tests for this logic (and maybe break it out to it's own function to make it easier to tests)